### ISSUE TITLE
breaks accepts_nested_attributes_for

### ISSUE THREAD

**Issue by amcox** [timlinquist/simple_eav#7]
I'm not sure if this is the way I've set up my code, or a bug with SimpleEav.

With a user model like this:

``` ruby
class User < ActiveRecord::Base

   include SimpleEav 
   configure_simple_eav :custom_fields # This sets up the serialization for your custom attributes

  devise :database_authenticatable,
         :recoverable, :rememberable, :trackable, :authentication_keys => [:username, :district_id]

  # Setup accessible (or protected) attributes for your model
  attr_accessible :email, :password, :password_confirmation, :remember_me, :username,
                  :first_name, :last_name, :position, :teacher, :district_id, :role_ids, :teachingassignments_attributes, :district_admin

  has_many :teachingassignments
  has_many :grades, :through => :teachingassignments
  has_many :subjects, :through => :teachingassignments
  has_many :jobs, :through  => :teachingassignments 

  accepts_nested_attributes_for :teachingassignments, :allow_destroy => true

...
```

And then form code like this:

``` rails
<%= form_for(@user) do |f| %>

    <div>
        <% f.fields_for :teachingassignments do |builder| %>  
        <%= render 'teachingassignment_fields', :f => builder %>  
      <% end %> 
        <p><%= link_to_add_fields "Add Position", f, :teachingassignments %></p>
    </div>
```

and

``` rails
<% if f.object.active? %>
    <div class="fields">
        <span class="field">
          <%= f.label :grade %>
            <%= f.collection_select( :grade_id, Grade.where(:district_id => @current_district.id).all, :id, :grade,
                                 {:prompt   => "Select a Grade"}) %>
        </span>
        <span class="field">
          <%= f.label :can_observe, "Can observe" %>
          <%= f.check_box :can_observe %>
        </span>
          <%= f.hidden_field :active, :checked => true %>
            <%= link_to_function "remove", "ta_remove_fields(this)" %>
    </div>
<% end %>
```

the teachingassignment attributes end up getting saved in the simple_eav column instead of as a new teaching assignment record. Disabling the simple_eav call and setup makes the records save properly.

Sorry to be submitting bugs without answers to them.

**Comment by timlinquist** [comment]
This is because you don't have a column defined for the nested attributes (as you shouldn't it's an association.)  SimpleEav thinks you want this to be a custom attribute on the User.

Give me a little bit of time to look into a fix for this.  NestedAttributes are part of AR proper so I'll figure out something.

**Comment by timlinquist** [comment]
Please let me know if this bug isn't fixed with latest version 0.5.0.  I'll reopen if you can still reproduce.
